src/PSPackageProject.psm1

```diff
@@ -93,6 +93,102 @@ function Invoke-PSPackageProjectTest
     )
 
     ## TODO implement calling tests
+    Invoke-BinSkim
+}
+
+function Invoke-BinSkim
+{
+    [CmdletBinding(DefaultParameterSetName='default')]
+    param(
+        [Parameter(ParameterSetName='byPath',Mandatory)]
+        [string]
+        $Path,
+        [Parameter(ParameterSetName='byPath')]
+        [string]
+        $Filter = '*'
+    )
+
+    $sourceName = 'Nuget'
+    Register-PackageSource -ProviderName NuGet -Name $sourceName -Location https://api.nuget.org/v3/index.json -erroraction ignore
+    $packageName = 'microsoft.codeanalysis.binskim'
+    $packageLocation = Join-Path -Path ([System.io.path]::GetTempPath()) -ChildPath 'pspackageproject-packages'
+    Write-Verbose "Finding binskim..." -Verbose
+    $packageInfo = Find-Package -Name $packageName -Source $sourceName
+    if($IsLinux)
+    {
+        $binaryName ='BinSkim'
+        $rid = 'linux-x64'
+    }
+    elseif($IsWindows)
+    {
+        $binaryName ='BinSkim.exe'
+        if([Environment]::Is64BitOperatingSystem)
+        {
+            $rid = 'win-x64'
+        }
+        else
+        {
+            $rid = 'win-x86'
+        }
+    }
+    else {
+        Write-Warning "unsupported platform"
+        return
+    }
+
+    $dirName = $packageInfo.Name + '.' + $packageInfo.Version
+    $toolLocation = Join-Path -Path $packageLocation -ChildPath $dirName -AdditionalChildPath 'tools', 'netcoreapp2.0', $rid, $binaryName
+    if(!(test-path -path $toolLocation))
+    {
+        Write-Verbose "Installing binskim..." -Verbose
+        $packageInfo | Install-Package -Destination $packageLocation -Force
+    }
+
+    if($IsLinux)
+    {
+        chmod a+x $toolLocation
+    }
+
+    if($Path)
+    {
+        $resolvedPath = (Resolve-Path -Path $Path).ProviderPath
+        $toAnalyze = Join-Path -Path $resolvedPath -ChildPath $Filter
+    }
+    else
+    {
+        $toAnalyze = Join-Path -Path $packageLocation -ChildPath $dirName -AdditionalChildPath 'tools', 'netcoreapp2.0', $rid, '*'
+    }
+
+    $outputPath =  Join-Path -Path ([System.io.path]::GetTempPath()) -ChildPath 'pspackageproject-results.json'
+    Write-Verbose "Running binskim..." -Verbose
+    & $toolLocation analyze $toAnalyze --output $outputPath --pretty-print  > binskim.log 2>&1
+
+    $testsPath =Join-Path -Path $PSScriptRoot -ChildPath 'tasks' -AdditionalChildPath 'BinSkim', 'binskim.tests.ps1'
+
+    Write-Verbose "Generating test results..." -Verbose
+    Invoke-Pester -Script $testsPath -OutputFile ./binskim-results.xml -OutputFormat NUnitXml
+    Publish-AzDevOpsArtifact -Path ./binskim-results.xml -Title 'BinSkim Results' -Type NUnit
+}
+
+function Publish-AzDevOpsArtifact
+{
+    param(
+        [parameter(Mandatory)]
+        [string]
+        $Path,
+        [parameter(Mandatory)]
+        [string]
+        $Title,
+        [string]
+        $Type = 'NUnit'
+    )
+
+    $artifactPath = (Resolve-Path $Path).ProviderPath
+
+    if($env:TF_BUILD)
```

Comment:
If this function only is supposed to work in Azure DevOps, should this be:

```
if (-not $env:TF_BUILD)
{
    Write-Warning "something about only working in Azure Devops"
}

# Content
```

Reply:
We just don’t do this if we aren’t on azure dev Ops.  no need for a warning.